quarkchain/cluster/simple_network.py

```diff
@@ -228,7 +232,7 @@ def get_connection_to_forward(self, metadata):
             await self.master_server.add_transaction(tx, self)
 
     async def handle_get_root_block_header_list_request(self, request):
-        if request.limit <= 0:
+        if request.limit <= 0 or request.limit > 2 * ROOT_BLOCK_HEADER_LIST_LIMIT:
```

Comment:
I think request.limit should be min(request.limit, ROOT_BLOCK_HEADER_LIST_LIMIT).  This means ROOT_BLOCK_HEADER_LIST_LIMIT won't be a parameter in consensus.

Reply:
we don't want ROOT_BLOCK_HEADER_LIST_LIMIT to be part of consensus, just a way to avoid DOS from evil peers